storage/innobase/fts/fts0que.cc

```diff
@@ -1557,6 +1572,10 @@ fts_merge_doc_ids(
 				query, ranking->doc_id, ranking->rank);
 
 		if (query->error != DB_SUCCESS) {
+                        if (query->oper == FTS_EXIST && query->intersection != NULL)
```

Comment:
The query->oper == FTS_EXIST check looks redundant: the function asserts query->intersection == NULL at the start and only sets it to something if that check passes